### PR TITLE
updating receive triggers

### PR DESCRIPTION
**Pull request by daedela**

### CODE REVIEW

File: src/dg_triggers.cpp [hunk 1]
```diff
@@ -408,26 +408,41 @@ void hitprcnt_mtrigger(CharData *ch) {
 int receive_mtrigger(CharData *ch, CharData *actor, ObjData *obj) {
     TrigData *t;
     char buf[MAX_INPUT_LENGTH];
+    int ret_val = 1;
+
+    std::string s = std::to_string(GET_OBJ_VNUM(obj));
+    const char *vnum = s.c_str();
```
[the remark this answers is not in the document]
Comment (daedela) [reply src/dg_triggers.cpp:414]: I don't really understand what's going on here, tbh.  I just know I need to get the string to check in the arg field.

File: src/dg_triggers.cpp [hunk 2]
```diff
@@ -408,26 +408,41 @@ void hitprcnt_mtrigger(CharData *ch) {
...
         !char_susceptible_to_triggers(ch))
-        return 1;
+        return ret_val;
 
     for (t = TRIGGERS(SCRIPT(ch)); t; t = t->next) {
         if (IS_SET(GET_TRIG_TYPE(t), MTRIG_RECEIVE)) {
+          /*
             if (GET_TRIG_DEPTH(t)) {
-                /* The receive trigger is currently executing */
+                /* The receive trigger is currently executing
```
[the remark this answers is not in the document]
Comment (daedela) [reply src/dg_triggers.cpp:424]: Okay.

File: src/dg_triggers.cpp [hunk 3]
```diff
@@ -408,26 +408,41 @@ void hitprcnt_mtrigger(CharData *ch) {
...
+                ret_val = 0;
+                break;
+            } else {
+              */
+
+                if (GET_TRIG_ARG(t) && word_check(vnum, GET_TRIG_ARG(t)) || 
+                  (!GET_TRIG_ARG(t) || !*GET_TRIG_ARG(t))) {
+
+                    if (random_number(1, 100) <= GET_TRIG_NARG(t)) {
+                        ADD_UID_VAR(buf, t, actor, "actor");
```
[the remark this answers is not in the document]
Comment (daedela) [reply src/dg_triggers.cpp:435]: How?

File: src/dg_triggers.cpp [hunk 4]
```diff
@@ -408,26 +408,41 @@ void hitprcnt_mtrigger(CharData *ch) {
...
+                        ADD_UID_VAR(buf, t, obj, "object");
+                        ret_val = script_driver(&ch, t, MOB_TRIGGER, TRIG_NEW);
```
[the remark this answers is not in the document]
Comment (daedela) [reply src/dg_triggers.cpp:437]: Happy to, I just don't know how!

File: src/dg_triggers.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (stridera) [reply src/dg_triggers.cpp:424]: If we don't want this anymore, we should just delete it.

File: src/dg_triggers.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (stridera) [reply src/dg_triggers.cpp:435]: Oh, we can totally do quantity now... just move the mreceive_trigger to the part where we're stepping through the objects sent.

File: src/dg_triggers.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (stridera) [reply src/dg_triggers.cpp:437]: Just add a ```
if (!ret_val)
    return ret_val```
That's assuming a 0 is a success.

[the remark this answers is not in the document]
Comment (daedela) [reply src/dg_triggers.cpp]: Done.  Let me know if this looks right.

File: src/dg_triggers.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (daedela) [reply src/dg_triggers.cpp]: If we move the trigger place though, how do we block objects from getting passed?

File: src/dg_triggers.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (daedela) [reply src/dg_triggers.cpp]: Done

**Review by stridera — APPROVED** [review]

**Pull request merged**